hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java

```diff
@@ -126,11 +130,25 @@ public boolean shouldThrottle(
      * @param queryFunction the functionality of the query
      * @param now the time at which the query is being processed
      * @param query the query to update the throttle requirements for
+     * @param queryPayerId the payer id of the query
      * @return whether the query should be throttled
      */
     public boolean shouldThrottle(
-            @NonNull final HederaFunctionality queryFunction, @NonNull final Instant now, @NonNull final Query query) {
+            @NonNull final HederaFunctionality queryFunction,
+            @NonNull final Instant now,
+            @NonNull final Query query,
+            @Nullable final AccountID queryPayerId) {
         final var configuration = configProvider.getConfiguration();
+
+        if (queryPayerId == null) {
+            return false;
+        }
```

Comment:
Is this okay? If not how can I get the **payerId** in the `QueryWorkflowImpl.handleQuery` without calling `ingestChecker.runAllChecks(...)`?

Reply:
`queryPayerId` is `null` if the query is free. There is a throttle defined for free queries. Thus, I think returning `false` is wrong. I guess we should only skip the `throttleExempt()`-check.

Reply:
Makes sense. Fixed.